### PR TITLE
Fix undefined variables

### PR DESCRIPTION
**Pull request by Danielhiversen**

### CODE REVIEW

File: keras/backend/cntk_backend.py [hunk 1]
```diff
@@ -2065,20 +2065,20 @@ def stop_gradient(variables):
...
-    ndim_expr = ndim(then_expression)
-    if ndim_cond > ndim_expr:
+    cond_ndim = ndim(condition)
+    expr_ndim = ndim(then_expression)
+    if cond_ndim > expr_ndim:
         raise ValueError('Rank of condition should be less'
                          ' than or equal to rank of then and'
                          ' else expressions. ndim(condition)=' +
                          str(cond_ndim) + ', ndim(then_expression)'
                          '=' + str(expr_ndim))
```
Comment (Danielhiversen) [inline keras/backend/cntk_backend.py]: `cond_ndim` and `expr_ndim` was undefined.
I chose to use `expr_ndim` instead `ndim_expr` to be consistent with tensorflow_backend.py and theano_backend.py

Comment (fchollet) [reply keras/backend/cntk_backend.py]: Please add a test checking that the ValueError is being raised correctly (`pytest.raises(ValueError)`)

File: keras/backend/cntk_backend.py [hunk 2]
```diff
@@ -2065,20 +2065,20 @@ def stop_gradient(variables):
...
-    elif ndim_cond < ndim_expr:
+    elif cond_ndim < expr_ndim:
```
Comment (fchollet) [inline keras/backend/cntk_backend.py]: Please also add a test checking this case `cond_ndim < expr_ndim`

**Comment by Danielhiversen** [comment]
@fchollet I do not understand why the test is failing.


Without  `pytest.raises(ValueError):` I get a valueError as expected:
https://travis-ci.org/fchollet/keras/jobs/301847000#L2148

If I add the ` with pytest.raises(ValueError):` to catch the valueError, no valueError is raised:
https://travis-ci.org/fchollet/keras/jobs/301859882#L2146

**Comment by fchollet** [comment]
@Danielhiversen I assume the `ValueError` is raised when running the tests with some backends, and not raised with other backends. Hence neither catching it or not catching it will work (some subset of the tests will always fail).

What are you getting *when running the tests locally* with each of the 3 backends?

**Pull request closed without merging**